File: autodroid-app/app/src/main/java/com/autodroid/proxy/viewmodel/AppViewModel.java
package com.autodroid.proxy.viewmodel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;
import java.util.Map;

public class AppViewModel extends ViewModel {
    // Server information
    private final MutableLiveData<String> serverIp = new MutableLiveData<>();
    private final MutableLiveData<Boolean> serverConnected = new MutableLiveData<>(false);
    private final MutableLiveData<Map<String, Object>> serverInfo = new MutableLiveData<>();
    
    // Workflows information
    private final MutableLiveData<List<Map<String, Object>>> availableWorkflows = new MutableLiveData<>();
    private final MutableLiveData<Map<String, Object>> selectedWorkflow = new MutableLiveData<>();
    
    // Test reports information
    private final MutableLiveData<List<Map<String, Object>>> testReports = new MutableLiveData<>();
    private final MutableLiveData<Map<String, Object>> selectedReport = new MutableLiveData<>();
    
    // Real-time execution information
    private final MutableLiveData<Map<String, Object>> currentExecution = new MutableLiveData<>();
    private final MutableLiveData<String> executionLog = new MutableLiveData<>();
    
    // Device information
    private final MutableLiveData<String> deviceIp = new MutableLiveData<>();
    
    // Getters for LiveData
    public MutableLiveData<String> getServerIp() {
        return serverIp;
    }
    
    public MutableLiveData<Boolean> getServerConnected() {
        return serverConnected;
    }
    
    public MutableLiveData<Map<String, Object>> getServerInfo() {
        return serverInfo;
    }
    
    public MutableLiveData<List<Map<String, Object>>> getAvailableWorkflows() {
        return availableWorkflows;
    }
    
    public MutableLiveData<Map<String, Object>> getSelectedWorkflow() {
        return selectedWorkflow;
    }
    
    public MutableLiveData<List<Map<String, Object>>> getTestReports() {
        return testReports;
    }
    
    public MutableLiveData<Map<String, Object>> getSelectedReport() {
        return selectedReport;
    }
    
    public MutableLiveData<Map<String, Object>> getCurrentExecution() {
        return currentExecution;
    }
    
    public MutableLiveData<String> getExecutionLog() {
        return executionLog;
    }
    
    public MutableLiveData<String> getDeviceIp() {
        return deviceIp;
    }
    
    // Setters
    public void setServerIp(String ip) {
        serverIp.setValue(ip);
    }
    
    public void setServerConnected(boolean connected) {
        serverConnected.setValue(connected);
    }
    
    public void setServerInfo(Map<String, Object> info) {
        serverInfo.setValue(info);
        // Extract server IP from info if available
        if (info != null && info.containsKey("ip")) {
            serverIp.setValue(info.get("ip").toString());
        }
    }
    
    public void setAvailableWorkflows(List<Map<String, Object>> workflows) {
        availableWorkflows.setValue(workflows);
    }
    
    public void setSelectedWorkflow(Map<String, Object> workflow) {
        selectedWorkflow.setValue(workflow);
    }
    
    public void setTestReports(List<Map<String, Object>> reports) {
        testReports.setValue(reports);
    }
    
    public void setSelectedReport(Map<String, Object> report) {
        selectedReport.setValue(report);
    }
    
    public void setCurrentExecution(Map<String, Object> execution) {
        currentExecution.setValue(execution);
    }
    
    public void appendExecutionLog(String logEntry) {
        String currentLog = executionLog.getValue() != null ? executionLog.getValue() : "";
        executionLog.setValue(currentLog + logEntry + "\n");
    }
    
    public void clearExecutionLog() {
        executionLog.setValue("");
    }
    
    public void setDeviceIp(String ip) {
        deviceIp.setValue(ip);
    }
}

File: autodroid-app/app/src/main/java/com/autodroid/proxy/viewmodel/AppViewModel.kt
package com.autodroid.proxy.viewmodel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class AppViewModel : ViewModel() {
    // Getters for LiveData
    // Server information
    val serverIp: MutableLiveData<String?> = MutableLiveData<String?>()
    val serverConnected: MutableLiveData<Boolean?> = MutableLiveData<Boolean?>(false)
    val serverInfo: MutableLiveData<MutableMap<String?, Any?>?> =
        MutableLiveData<MutableMap<String?, Any?>?>()

    // Workflows information
    val availableWorkflows: MutableLiveData<MutableList<MutableMap<String?, Any?>?>?> =
        MutableLiveData<MutableList<MutableMap<String?, Any?>?>?>()
    val selectedWorkflow: MutableLiveData<MutableMap<String?, Any?>?> =
        MutableLiveData<MutableMap<String?, Any?>?>()

    // Test reports information
    val testReports: MutableLiveData<MutableList<MutableMap<String?, Any?>?>?> =
        MutableLiveData<MutableList<MutableMap<String?, Any?>?>?>()
    val selectedReport: MutableLiveData<MutableMap<String?, Any?>?> =
        MutableLiveData<MutableMap<String?, Any?>?>()

    // Real-time execution information
    val currentExecution: MutableLiveData<MutableMap<String?, Any?>?> =
        MutableLiveData<MutableMap<String?, Any?>?>()
    val executionLog: MutableLiveData<String?> = MutableLiveData<String?>()

    // Device information
    val deviceIp: MutableLiveData<String?> = MutableLiveData<String?>()

    // Setters
    fun setServerIp(ip: String?) {
        serverIp.setValue(ip)
    }

    fun setServerConnected(connected: Boolean) {
        serverConnected.setValue(connected)
    }

    fun setServerInfo(info: MutableMap<String?, Any?>?) {
        serverInfo.setValue(info)
        // Extract server IP from info if available
        if (info != null && info.containsKey("ip")) {
            serverIp.setValue(info.get("ip").toString())
        }
    }

    fun setAvailableWorkflows(workflows: MutableList<MutableMap<String?, Any?>?>?) {
        availableWorkflows.setValue(workflows)
    }

    fun setSelectedWorkflow(workflow: MutableMap<String?, Any?>?) {
        selectedWorkflow.setValue(workflow)
    }

    fun setTestReports(reports: MutableList<MutableMap<String?, Any?>?>?) {
        testReports.setValue(reports)
    }

    fun setSelectedReport(report: MutableMap<String?, Any?>?) {
        selectedReport.setValue(report)
    }

    fun setCurrentExecution(execution: MutableMap<String?, Any?>?) {
        currentExecution.setValue(execution)
    }

    fun appendExecutionLog(logEntry: String?) {
        val currentLog = if (executionLog.getValue() != null) executionLog.getValue() else ""
        executionLog.setValue(currentLog + logEntry + "\n")
    }

    fun clearExecutionLog() {
        executionLog.setValue("")
    }

    fun setDeviceIp(ip: String?) {
        deviceIp.setValue(ip)
    }
}
